File: app/src/main/java/no/hiof/snailey/familyplaner/model/todo.java
package no.hiof.snailey.familyplaner.model;

import com.google.firebase.firestore.Exclude;

public class todo {

    @Exclude
    private String userID;
    private String title;
    private String description;
    private boolean active;
    private String family;

    public todo() {};

    public todo(String title, String description) {
        this.title = title;
        this.description = description;
    }

    public todo(String title, String description, boolean active, String family) {
        this.title = title;
        this.description = description;
        this.active = active;
        this.family = family;
    }

    @Exclude
    public String getUserID() {
        return userID;
    }

    public void setUserID(String userID) {
        this.userID = userID;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getFamily() {
        return family;
    }

    public void setFamily(String family) {
        this.family = family;
    }
}

File: app/src/main/java/no/hiof/snailey/familyplaner/data/ToDo.kt
package no.hiof.snailey.familyplaner.data

import com.google.firebase.database.Exclude


data class ToDo(
    @get:Exclude
    var id: String? = null,
    var name: String? = null,
    @get:Exclude
    var isDeleted: Boolean = false
) {
    override fun equals(other: Any?): Boolean {
        return if (other is ToDo) {
            other.id == id
        } else false
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        return result
    }

}